magento/magento2
# Fix "Syntax error or access violation:  Duplicate key name" If we app…

…ly fulltext index and unique constraint on same column.





### Description (*)
I have fixed the issue where applying a full-text index and a unique constraint on the same column was causing an error in Magento. The problem has been resolved, and this pull request includes the necessary changes to make this combination work seamlessly.

### Error Detail:
Schema File:
<img width="883" alt="image" src="https://github.com/magento/magento2/assets/74697541/c7080a2b-6006-4fd5-8c7d-6d4263c2a2e5">

Error:
<img width="710" alt="image" src="https://github.com/magento/magento2/assets/74697541/1feca2a9-8218-4d33-bbba-27ec701c3042">

### Manual testing scenarios (*)

1. Apply a full-text index and a unique constraint to the same column in a Magento database.
2. Verify that the operation completes successfully without errors.
3. Insert duplicate and non-duplicate values into the column.
4. Ensure that duplicate values trigger a unique constraint violation error, while non-duplicate values are allowed.
5. Confirm that the full-text indexing functionality still works as expected for the column.

### Questions or comments


### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
 - [*] All commits are accompanied by meaningful commit messages
